routes.jsx

```diff
@@ -15,21 +16,38 @@ import Navbar from './components/navbar/navbar.jsx';
 import Footer from './components/footer/footer.jsx';
 
 const App = React.createClass({
+  getInitialState() {
+    return {
+      suppressSplashScreen: localstorage.getItem(`suppressSplashScreen`) === `true`
+    };
+  },
   componentDidMount() {
-    setTimeout(function() {
-      this.splash.classList.add(`dismissed`);
-      document.querySelector(`#app`).classList.add(`splash-dismissed`);
-    }, 3000);
+    if (this.state.suppressSplashScreen !== `true`) {
+      setTimeout(function() {
+        this.splash.classList.add(`dismissed`);
+        document.querySelector(`#app`).classList.add(`splash-dismissed`);
+      }, 3000);
+      localstorage.setItem(`suppressSplashScreen`, `true`);
```

Comment:
@Pomax do you mind explaining more about 

> because we're working in React, I'm also going to say "let's not define functions on the fly".

I'd love to know why it's not a good idea to do that in React 😉